app/javascript/components/city-name.jsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { Query } from 'react-apollo';
+import gql from 'graphql-tag';
+import { WhiteTitleLarge } from '@components/typography';
+import { LoadingMessage, ErrorMessage } from '@messages/default-messages';
+
+const getCityName = gql`
+  {
+    primaryLocation {
+      cityName
+    }
+  }
+`;
+
+export const City = () => (
+  <Query query={getCityName}>
+    {({ loading, error, data }) => {
```

Comment:
```suggestion
    {({ loading, error, data: { primaryLocation: { cityName } }) => {
```

Nitpick: Since the only thing we're using from `data` is the city name, I would consider destructuring to it directly over the assignment on line 26.

Reply:
Interesting, what do you mean by destructing in this case? would I have to change the variable name on line 31?

Reply:
Okay, so the `({ loading, error, data })` syntax is called "destructuring". So is the `{ cityName }` in `const { cityName } = data.primaryLocation`. It's a relatively new syntax in JS, the old equivalent would look like

Reply:
Okay cool. Yeah its not a syntax I am familiar with. What would the old equivalent look like?

Reply:
Hm, I seem to be getting type errors where the "right side of assignment cannot be destructured." Even if I change the line `return cityName && <WhiteTitleLarge>{cityName}</WhiteTitleLarge>;` to `return data && <WhiteTitleLarge>{data}</WhiteTitleLarge>;`, I get the same issue.